platform-core-module/src/main/java/com/softicar/platform/core/module/program/AGProgram.java

```diff
@@ -106,20 +162,20 @@ public static AGProgram loadOrInsert(AGUuid programUuid) {
 
 	private static AGProgram executeLoadOrInsert(AGUuid programUuid) {
 
-		return AGProgram.TABLE//
-			.createSelect(SqlSelectLock.FOR_UPDATE)
```

Comment:
This eager `for-update` conditional lock was actually important: If we simply use `AGProgram.loadByProgramUuid` instead, multiple threads could try to concurrently insert `AGProgram` records with the same UUIDs.

This needs to be fixed/reverted.

Reply:
Ok, thanks for info! I reverted it.